src/Azure.Functions.Sdk/Targets/Azure.Functions.Sdk.targets

```diff
@@ -23,4 +26,20 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   <Import Project="$(MSBuildFunctionsTargetsPath)Microsoft.Azure.Functions.Designtime.targets"
           Condition="Exists('$(MSBuildFunctionsTargetsPath)Microsoft.Azure.Functions.Designtime.targets')" />
 
+  <Target Name="_ValidateFunctionsVersion">
+    <ItemGroup>
+      <_AllowedFunctionVersion Include="v4" InSupport="true" />
```

Comment:
I think this is fine because v3 and below are out of support; is the expectation that if they want to use this new SDK model they need to upgrade, or they should stay on the old SDK model?

Reply:
Kind of the same question about .NET 6/7 (less than 8)